example.ts

```diff
@@ -0,0 +1,27 @@
+
+import { KnownDevices } from "puppeteer";
+import { CollectorOptions, collect } from "./src";
+import { join } from 'path';
+
+(async () => {
+    const URL = 'example.com';
+    const EMULATE_DEVICE = 'iPhone 13 Mini';
+
+    const config: CollectorOptions = {
+        numPages: 3,
+        headless: false,
+        //emulateDevice: KnownDevices[EMULATE_DEVICE],
+        // Uncomment to run with desktop/laptop browser
+        // emulateDevice: {
+        //     viewport: {height: 1440, width: 800}, 
+        //     userAgent: "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36"
+        // },
+        outDir: join(__dirname, 'demo-dir'),
+    };
+
+    console.log(`Beginning scan of ${URL}`);
+
+    await collect(`http://${URL}`, config);
+
+    console.log(`Scan complete: ${config.outDir}`);
+})();
```

Comment:
I don't think we need the wrapping function anymore.
```suggestion
import { CollectorOptions, collect } from "./src";
import { join } from 'path';

const URL = 'example.com';
const EMULATE_DEVICE = 'iPhone 13 Mini';

const config: CollectorOptions = {
    numPages: 3,
    headless: false,
    //emulateDevice: KnownDevices[EMULATE_DEVICE],
    // Uncomment to run with desktop/laptop browser
    // emulateDevice: {
    //     viewport: {height: 1440, width: 800}, 
    //     userAgent: "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36"
    // },
    outDir: join(__dirname, 'demo-dir'),
};

console.log(`Beginning scan of ${URL}`);

await collect(`http://${URL}`, config);

console.log(`Scan complete: ${config.outDir}`);
```

Reply:
Great diff GitHub

Reply:
We do, collect() is still async and can't run outside an async function. Unless I'm missing something.